db/src/models/domain_actions.rs

```diff
@@ -256,7 +256,7 @@ impl NewDomainAction {
     }
 
     pub fn schedule_at(&mut self, schedule_at: NaiveDateTime) {
-        let diff = self.scheduled_at - schedule_at;
+        let diff = schedule_at - self.scheduled_at;
```

Comment:
Do we always know which is larger?
```suggestion
        let diff = schedule_at - self.scheduled_at;
        let diff = diff.abs();
```

Reply:
No, this is intentional. You may bring the scheduled time forward